source/Calamari.Shared/Integration/Packages/NuGet/InternalNuGetPackageDownloader.cs

```diff
@@ -94,7 +97,8 @@ private void DownloadPackageAction(string packageId, IVersion version, Uri feedU
             // NuGet V3 feed 
             else if (IsHttp(feedUri.ToString()) && feedUri.ToString().EndsWith(".json", StringComparison.OrdinalIgnoreCase))
             {
-                NuGetV3Downloader.DownloadPackage(packageId, version, feedUri, feedCredentials, targetFilePath);
+                var timeout = GetHttpTimeout();
```

Comment:
Consider pulling this out of the #if, and log a warning if it's set but not going to be used.

Reply:
Probably worth making this work for `NuGetV2Downloader.DownloadPackage` too

Reply:
> 
> 
> Probably worth making this work for `NuGetV2Downloader.DownloadPackage` too

This is harder than it looked. `NuGetV2Downloader` uses `HttpClient` to make HTTP requests, but it's not the built-in .net one, it's nuget's own version of it, which doesn't expose a timeout.

Reply:
Done. 👍